File: Code_plus/foundaional_algorithm/stack/word/word.cpp
#include <iostream>
#include <stack>
#include <string>
#include <string.h>


void see_all_stack(std::stack<char> s)
{
	while(!(s.empty()))
	{
		std::cout << s.top();
		s.pop();
	}
}
int main()
{
	std::stack <char> s;
	std::string input;
	int i;
	char dil;

	dil = ' ';
	getline(std::cin, input);
	i = 0;

	while(i < (int)input.length())
	{
		s.push(input[i]);
		if (input[i] == dil)
		{
			see_all_stack(s);
		}


		i++;
	}
}

File: Code_plus/foundaional_algorithm/stack/word/word.py
def main():
	command_times = int(input())

	j = 0;

	while(j < command_times):

		input_str = list(input())
		stack = list()
		output_stack = list()

		input_str.append(' ')
		i = 0
		while(i < len(input_str)):

			stack.append(input_str[i])
			if(input_str[i] == ' '):

				while(len(stack) != 0):
					output_stack.append(stack.pop())

			i+=1

		del output_stack[0]
		print(''.join(output_stack));
		j+=1

if __name__ == "__main__":
	main()
